src/interactivity/win32/Clipboard.cpp

```diff
@@ -203,40 +203,31 @@ void Clipboard::StoreSelectionToClipboard(bool const fAlsoCopyFormatting)
 
     // read selection area.
     const auto selectionRects = selection.GetSelectionRects();
-    const bool lineSelection = Selection::Instance().IsLineSelection();
 
     const auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
-    const auto& screenInfo = gci.GetActiveOutputBuffer();
-
-    const auto text = RetrieveTextFromBuffer(screenInfo,
-                                             lineSelection,
-                                             selectionRects);
-
-    CopyTextToSystemClipboard(text, fAlsoCopyFormatting);
-}
-
-// Routine Description:
-// - Retrieves the text data from the selected region of the text buffer
-// Arguments:
-// - screenInfo - what is rendered on the screen
-// - lineSelection - true if entire line is being selected. False otherwise (box selection)
-// - selectionRects - the selection regions from which the data will be extracted from the buffer
-TextBuffer::TextAndColor Clipboard::RetrieveTextFromBuffer(const SCREEN_INFORMATION& screenInfo,
-                                                           const bool lineSelection,
-                                                           const std::vector<SMALL_RECT>& selectionRects)
-{
-    const auto& buffer = screenInfo.GetTextBuffer();
-    const bool trimTrailingWhitespace = !WI_IsFlagSet(GetKeyState(VK_SHIFT), KEY_PRESSED);
-    const auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+    const auto& buffer = gci.GetActiveOutputBuffer().GetTextBuffer();
 
     std::function<COLORREF(TextAttribute&)> GetForegroundColor = std::bind(&CONSOLE_INFORMATION::LookupForegroundColor, &gci, std::placeholders::_1);
     std::function<COLORREF(TextAttribute&)> GetBackgroundColor = std::bind(&CONSOLE_INFORMATION::LookupBackgroundColor, &gci, std::placeholders::_1);
 
-    return buffer.GetTextForClipboard(lineSelection,
-                                      trimTrailingWhitespace,
-                                      selectionRects,
-                                      GetForegroundColor,
-                                      GetBackgroundColor);
+    bool includeCRLF, trimTrailingWhitespace;
+    if (WI_IsFlagSet(GetKeyState(VK_SHIFT), KEY_PRESSED))
+    {
+        // When shift is held, put everything in one line
+        includeCRLF = trimTrailingWhitespace = false;
+    }
+    else
+    {
+        includeCRLF = trimTrailingWhitespace = true;
+    }
+
+    const auto text = buffer.GetText(includeCRLF,
+                                     /*trimTrailingWhitespace*/ true,

```

Comment:
This is wacky. You set two variables to the same thing and then use one of them and put the true constant into the other.